extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/exec/ControllerImpl.java

```diff
@@ -1807,10 +1809,26 @@ private static MSQStatusReport makeStatusReport(
       @Nullable final MSQErrorReport errorReport,
       final Queue<MSQErrorReport> errorReports,
       @Nullable final DateTime queryStartTime,
-      final long queryDuration
+      final long queryDuration,
+      MSQWorkerTaskLauncher taskLauncher
   )
   {
-    return new MSQStatusReport(taskState, errorReport, errorReports, queryStartTime, queryDuration);
+    int pendingTasks = 0;
```

Comment:
Should pending tasks be 0 here or should it be the total number of tasks that the user has launched the query with? In case that is not known at this point here, we should probably change it to something like -1 to denote unknown.

Reply:
I have set it to -1, ideally, it should not be null and the issue would be fixed by the next status report.